Lib/venv/__init__.py

```diff
@@ -116,7 +116,7 @@ def create_if_needed(d):
             elif os.path.islink(d) or os.path.isfile(d):
                 raise ValueError('Unable to create directory %r' % d)
 
-        if os.pathsep in env_dir:
+        if isinstance(env_dir, str) and os.pathsep in env_dir:
```

Comment:
It looks like the intent here is to enforce that env_dir is only a single path segment. Should we preserve that behavior for path-likes too? For example, we could call `env_dir = os.fspath(env_dir)` first.

I'm not sure why we need to check for this at all though. What is the reason we have this check?

Reply:
It's checking for `os.pathsep`, not `os.sep`: it was a fix for bpo-43218. I agree, we should retain the behaviour for path-likes too - otherwise the problem originally solved for will come back.